### PR TITLE
feature: URL prefix case-sensitive

### PR DESCRIPTION
**Pull request by GLYASAI**
### Issues associated with this PR

#1633

### Solutions

make the configuration 'case_sensitive' work.

### CODE REVIEW

**Review by antJack — APPROVED** [review]
LGTM

File: pkg/filter/stream/jwtauthn/matcher.go [hunk 1]
```diff
@@ -69,7 +70,11 @@ func newPrefixMatcher(rule *jwtauthnv3.RequirementRule) Matcher {
 }
 
 func (p *prefixMatcher) Matches(headers api.HeaderMap, requestPath string) bool {
-	if p.baseMatcher.matchRoutes(headers, requestPath) && strings.HasPrefix(requestPath, p.prefix) {
+	prefixMatch := strings.HasPrefix(requestPath, p.prefix)
+	if !p.baseMatcher.caseSensitive {
```
Comment (nejisama) [inline pkg/filter/stream/jwtauthn/matcher.go:74]: 这里是不是有重复计算。 先判断caseSensitive？

Comment (GLYASAI) [reply pkg/filter/stream/jwtauthn/matcher.go:74]: ok

**Review by nejisama — APPROVED** [review]

**Review by antJack — APPROVED** [review]
LGTM

**Pull request merged**